src/plot_api/plot_api.js

```diff
@@ -2533,7 +2533,7 @@ Plotly.relayout = function relayout(gd, astr, val) {
  * @param {string id or DOM element} gd
  *      the id or DOM element of the graph container div
  */
-Plotly.transition = function(gd, data, layout, traceIndices, transitionConfig) {
+Plotly.transition = function(gd, data, layout, traceIndices, transitionConfig, onTransitioned) {
```

Comment:
Short answer: because we need confirmation from the transition when it has completed, but it's a bit too complicated right at the moment to hook up the event dispatcher and attach and detach the events in sequence.